rbc/irtools.py

```diff
@@ -438,6 +439,27 @@ def compile_to_LLVM(functions_and_signatures,
       LLVM module instance. To get the IR string, use `str(module)`.
 
     """
+    # check LLVM version before compiling to LLVM
+    server_llvm_version = target_info.llvm_version
+    client_llvm_version = llvm.llvm_version_info
+
+    if (server_llvm_version[0], client_llvm_version[0]) == (11, 14):
+        c_llvm = '.'.join(map(str, client_llvm_version))
+        s_llvm = '.'.join(map(str, server_llvm_version))
+        flag = 'RBC_DISABLE_LLVM_MISMATCH_ERROR'
+        msg = (f'The client LLVM version ({c_llvm}) is greater than the server '
+               f'LLVM version ({s_llvm}). This is known to be unsupported. '
+               'Please, downgrade to a previous release of Numba that uses the '
+               'same LLVM version as the HeavyDB server. For more information, '
+               'see the table below:\n\n'
+               'https://github.com/numba/llvmlite#compatibility\n'
+               'https://github.com/heavyai/heavydb#dependencies\n\n'
+               f'To disable this error, run RBC with {flag}=1 flag enabled.')
+
+        DISABLE_LLVM_MISMATCH_ERROR = int(os.environ.get(flag, False))
+        if not DISABLE_LLVM_MISMATCH_ERROR:
+            raise LLVMVersionMismatchError(msg)
```

Comment:
Would this be the most appropriated place to put this check?

Reply:
Okay, I can change that. We only check for the LLVM 11/14 combination, which is known to be problematic.